ipfs-gateway-finder/src/main.rs

```diff
@@ -141,11 +141,14 @@ async fn do_probing() -> Result<()> {
 
     let monitor_api_address = matches.value_of("monitor_api_address").unwrap();
     info!("using monitoring node API at {}", monitor_api_address);
-    let u = http::uri::Builder::new()
-        .scheme(http::uri::Scheme::HTTP)
-        .authority(monitor_api_address)
-        .path_and_query("/")
-        .build()
+    
+    let monitor_api_address = if monitor_api_address.starts_with("http://") {
+        monitor_api_address.to_string()
+    } else {
+        format!("http://{}", monitor_api_address)
+    };
+    
+    let u = monitor_api_address.parse::<http::Uri>()
```

Comment:
Does this whole construction work? I don't remember if the `path_and_query` in the original was important... probably to strip any path components already present? But should be fine like this.